### ISSUE TITLE
Cursor persists on previous line with new OpenGL rendering engine

### ISSUE THREAD

**Issue by DesiQuintans** [rstudio/rstudio#4390]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1303
    OS Version      : Windows 7 64-bit
    R Version       : 3.5.2

### Steps to reproduce the problem

1. Check that the rendering settings are at their default settings (below). In Windows, this will make RStudio try to use the desktop OpenGL renderer.

![image](https://user-images.githubusercontent.com/3200601/53690190-dbf4ac00-3db9-11e9-9518-4412672fdb34.png)

2. Create a new R script file with this example text:

```
a 
b 
c 
d 
```

3. Place the cursor on any line. Notice that there is a white dot blinking about 2x faster than the cursor, located 1 character behind the cursor's current location. If the cursor is at the start of the line, it will even draw itself on top of the gutter.
4. Use the arrow keys to move the cursor between lines. The dot stays in the same place.
5. Use the mouse to put the cursor on a different line. The dot will move itself _to the line where the cursor used to be_.
6. Try changing the rendering options by enablind and disabling the three rendering checkboxes in any combination that you like. The problem will persist.
7. Change rendering to force Software rendering, and the problem will go away.

### Describe the problem in detail

![cursor bug](https://user-images.githubusercontent.com/3200601/53690185-bd8eb080-3db9-11e9-8fd2-aaabb732ec39.gif)

Interestingly, my cursor is set to a lilac colour by my RStudio theme but the blinking dot is pure white (#FFFFFF), and I don't use that colour anywhere in my theme. The only place it appears in the theme is in the baked definitions for terminal colours.

```
.xtermColor231 { color: #ffffff !important; }
.xtermBgColor231 { background-color: #ffffff; }
```

### Describe the behavior you expected

No part of the cursor should not persist on previous lines.

**Comment by grawil** [comment]
This has been an issue for me too since at least as far back as mid-Jan (1.2.1239, maybe earlier). I am also using Windows 7 64-bit.

**Comment by kevinushey** [comment]
Not to be a downer, but Windows 7 is now nearly a 10 year old operating system, and is less than a year away from [end of support](https://www.microsoft.com/en-us/windowsforbusiness/end-of-windows-7-support) from Microsoft.

If the issue occurs in Windows 7 but not Windows 10, then it's unlikely that we'll fix it.

**Comment by DesiQuintans** [comment]
Hi @kevinushey, thanks for prompting that. I am able to reproduce the bug on my university machine:

```
RStudio Edition : Desktop
RStudio Version : 1.2.1322-2
OS Version      : Windows 10 Enterprise 64-bit
R Version       : 3.5.2
```

The way the bug presents itself on this platform is different. Instead of going to the last position of the cursor, the flashing dot is now stuck on the cursor position of the last typed text. Clicking the cursor to a new position no longer moves the dot, it's now only moved by typing new input.

![cursor dot](https://user-images.githubusercontent.com/3200601/54156806-1dace300-449b-11e9-9839-5fbd43f9dd04.gif)

Windows 10 also seems to not activate the OpenGL renderer when it's set to auto-detect? This only happens when the renderer is specifically set to OpenGL. Whether the _GPU blacklist_ and _driver bug workarounds_ options are enabled or disabled makes no difference.

**Comment by kevinushey** [comment]
Interesting; thanks.

We disable the GPU by default for these integrated Intel graphics cards as a number of users have reported issues with them in the past when enabled. See e.g.

https://github.com/rstudio/rstudio/blob/b2e28db767ca913b1ad0587e6441fd80dc9ca4f2/src/cpp/desktop/DesktopMain.cpp#L284-L323

Are your graphics card drivers up-to-date? If not, it's possible that updating them would resolve these sorts of issues.

**Comment by DesiQuintans** [comment]
I'm not sure if I have the auth to update them, but I can have a look tomorrow.

I used RStudio all day today with the OpenGL renderer enabled and found that the blinking dot is not just limited to one document, but exists on the canvas of the editor or something, because it stays in the same place on the screen when I change to a different document, or even when I look at the data viewer.

**Comment by AshesITR** [comment]
I can reproduce the issue on a GTX 1080Ti on Win7 x64 with driver version 419.35 (as of writing this, the most recent one).

**Comment by dicorynia** [comment]
Hi.
Similar problem (RStudio v. 1.2.1327 on windows 7 64bit, Nvidia GTX960 v. 378.92). The blinking dot follows the cursor, in the console or the editor.
No blinking dot if I switch to software rendering in the RStugio options.

Thanks

**Comment by lz1nwm** [comment]
Just to confirm the bug with the cursor - RStudio v. 1.2.1335 on Win7 64bit, Radeon 7350. No blinking dot if I switch to software rendering, as well.

**Comment by FluffierThanThou** [comment]
As reported in #4822, I have the same problem with RStudio 1.2.1335, Win10 Enterprise 1703 (x64) and an Intel HD 4600 integrated graphics chip using (fairly) recent drivers. The dot only appears when using OpenGL rendering.

**Comment by brupelo** [comment]
Hello guys, this bug has been reported for a while at [https://bugreports.qt.io/browse/QTBUG-74492](https://bugreports.qt.io/browse/QTBUG-74492). So if you've got new info/insights that could help Qt developers to narrow it down, please report there. Really annoying and nasty this bug... We've already provided couple of mcves there though.

Thanks.

**Comment by Spaak** [comment]
I'm having the same issue on RStudio 1.2.1335, Windows 10 Enterprise 2016 LTSB 10.0.14393 Build 14393 (x64). Graphics card is an NVIDIA Quadro K600, driver version 369.39 (DirectX 12).

As described above, the issue does **not** occur when I select software rendering.

**Comment by kevinushey** [comment]
Evidently, this issue will be fixed in the next version of Qt (5.12.5) but it feels frightening to take yet another Qt update in this release cycle.

@jmcphers do you think it's worth taking an update for this bug, or deferring for v1.3, or something else?

**Comment by jmcphers** [comment]
I don't think we should patch 1.2 for this. Very few people are affected (apologies to those in this thread who are!) and historically Qt 5.12 updates have been very disruptive as they are prone to changing not only functionality but dependencies in their point releases. :-/ We should take a Qt update for 1.3 instead.

**Comment by JeffreyBlumenthal** [comment]
I fixed this problem by toggling Rendering Engine to "Software", (Tools -> Global ->General [Advanced tab] per this: 

https://support.rstudio.com/hc/en-us/articles/360017886674-Troubleshooting-RStudio-Rendering-Errors

**Comment by brunoappentra** [comment]
After a lot of effort we'd fixed this issue a while ago. For more info just check https://bugreports.qt.io/browse/QTBUG-74492 . In case you want a proper fix you'll just need to upgrade this repo to use the proper version of Qt.

**Comment by kevinushey** [comment]
Presumedly this issue is fixed already given that we've updated to Qt 5.12.8 on Windows; I'm going to close as fixed but please re-open if you see it again.

(I also doubt we'll see it in the Electron port.)
